### PR TITLE
Fix/prettier should run in every folder, not just in editor

### LINKED ISSUE #988: Make prettier work everywhere (performance-test too)

### PR DESCRIPTION
**Pull request by enidemi**
Fixes #988 

**Problem:**
The pre-commit hook only watches for changes editor/src. The existing hook is including utopia-api, but it doesn't do anything now (maybe this was working in the past?).

**Fix:**
Adding the pre-commit hook to the root folder allows it to check in all places.

**Commit Details:**
- updated husky to the latest version
- extended `lint-staged` to run prettier on editor, utopia-api, utopia-vscode-common, performance-test directories
- change `prettier-check` script to include these new directories
- ran prettier once to make the existing code pretty, this effected utopia-vscode-common

### CODE REVIEW

**Review by Rheeseyb — COMMENTED** [review]
This needs an update to `shell.nix` to ensure `npm install` is being called at the root level as part of all developer workflows. It's probably easiest to add this as part of either `start-full` or `install-editor`

**Review by balazsbajorics — APPROVED** [review]

File: shell.nix [hunk 1]
```diff
@@ -23,6 +23,7 @@ let
     (pkgs.writeScriptBin "install-editor" ''
       #!/usr/bin/env bash
       set -e
+      ${node}/bin/npm --scripts-prepend-node-path=true install
```
Comment (Rheeseyb) [inline shell.nix:26]:
```suggestion
      cd $(${pkgs.git}/bin/git rev-parse --show-toplevel)
      ${node}/bin/npm --scripts-prepend-node-path=true install
```
I'm pretty sure we'd need to `cd` here to ensure we're in the right directory first

**Review by Rheeseyb — APPROVED** [review]
With the proposed `cd` this looks good to me

File: shell.nix (hunk 1 above)
Comment (enidemi) [reply shell.nix:27]: thanks! fixed :)

**Pull request merged**